source/Session.h

```diff
@@ -156,6 +156,12 @@ class SessionConfig : public FpsConfig {
 	Any toAny(const bool forceAll = false) const;
 	float getTrialsPerBlock(void) const;			// Get the total number of trials in this session
 	Array<String> getUniqueTargetIds() const;
+
+	void respawnPlayer() { player.respawnToPos = true; }
```

Comment:
I strongly suggest against adding this method here unless there is a good reason for it. You should be able to add a method like this to the `PlayerEntity` or just call the `PlayerEntity::respawn()`. Having a session-level method that duplicates this/sets a semaphore seems odd.

More importantly I don't see any added use of `respawnToPos` I assume this was added in a previous commit?

Reply:
For some reason, player entity gets initialized later than the gui, I could not figure out a way to pass the actual player entitiy to gui. That's why I went with this. 
respawnToPos is checked in the player entity onSimulation, and whenever it is true, respawn method is called from there.

Reply:
Can you use a callback method that performs a `notNull()` check on the `PlayerEntity` prior to calling respawn? Using a `PlayerConfig` parameter as a semaphore isn't recommended.

Reply:
Can you give me an example of such a method? Also, the player entity gets saved to the local variable of gui while creating the gui. I didn't find any easy way to update afterward when the player is actually in the scene.